### ISSUE TITLE
Alphabetic ordering of patches-search files

### ISSUE THREAD

**Issue by hnsr** [vaimo/composer-patches#29]
In our TravisCI environment, it seems that vaimo/composer-patches are applied in random order, instead of alphabetic order.

I have multiple patches for a package, named as follows:

```
firebear-configurableproducts-1-fix-order-items.patch
firebear-configurableproducts-2-fix-qty-check.patch
firebear-configurableproducts-3-fix-console-command-early-db-access.patch
firebear-configurableproducts-4-fix-stock-indexer-query.patch
```
They are stored under the root package folder under `composer-patches`, and configured in composer.json through:

```
    "extra": {
        "patches-search": "composer-patches"
    },
```

In my local dev environment these are applied in alphabetic order; however in our TravisCI environment, patch 3 and 4 are applied first, which fails as patch 4 is dependent on patch 1. 

Although you can enforce patch ordering with the @after keyword, it would be nice if alphabetic ordering could be enforced in some way, so we can just drop in patches with the right filename and be on our way.

I suppose making alphabetic sorting the default might break a lot of projects if they haven't used the @after keyword and patches by chance apply correctly, so alphabetic sorting should probably be made a toggle

**Comment by allanpaiste** [comment]
Fixed in 3.47.1 & 4.16.1;

Patches (when automatically detected from FS) now sorted alphanumerically.

**Comment by allanpaiste** [comment]
Fixed in 3.47.1 & 4.16.1;

Patches (when automatically detected from FS) now sorted alphanumerically.

**Comment by hnsr** [comment]
A bit late, but thanks!
